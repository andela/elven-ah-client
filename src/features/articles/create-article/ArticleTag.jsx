import React from 'react';
import CreatableSelect from 'react-select/lib/Creatable';
import PropTypes from 'prop-types';

const ArticleTag = ({ handleChange, tags }) => (
  <CreatableSelect
    isClearable
    isMulti
    onChange={handleChange}
    placeholder="Tag your story"
    value={tags}
  />
);

ArticleTag.propTypes = {
  tags: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  handleChange: PropTypes.func.isRequired,
};

export default ArticleTag;
