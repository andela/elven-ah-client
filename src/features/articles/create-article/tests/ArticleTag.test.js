import React from 'react';
import { shallow } from 'enzyme';
import CreatableSelect from 'react-select/lib/Creatable';
import ArticleTag from '../ArticleTag';

const tags = [
  { value: 'jest', label: 'jest' },
  { value: 'react', label: 'react' },
  { value: 'enzyme', label: 'enzyme' },
];

describe('Article Tag component', () => {
  it('should render without errors', () => {
    const wrapper = shallow(<ArticleTag
      tags={tags}
      handleChange={() => jest.fn()}
    />);
    expect(wrapper.find(CreatableSelect)).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
});
