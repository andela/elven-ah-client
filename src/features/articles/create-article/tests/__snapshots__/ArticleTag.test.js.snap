// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Tag component should render without errors 1`] = `
<StateManager
  defaultInputValue=""
  defaultMenuIsOpen={false}
  defaultValue={null}
  isClearable={true}
  isMulti={true}
  onChange={[Function]}
  placeholder="Tag your story"
  value={
    Array [
      Object {
        "label": "jest",
        "value": "jest",
      },
      Object {
        "label": "react",
        "value": "react",
      },
      Object {
        "label": "enzyme",
        "value": "enzyme",
      },
    ]
  }
/>
`;
